lib/src/arithmetic/compound_adder.dart

```diff
@@ -68,27 +77,55 @@ class CarrySelectCompoundAdder extends CompoundAdder {
     return splitData;
   }
 
+  /// General width splitter
+  static List<int> _splitAdderNBitFunctor(int adderWidth, int n) {
+    final blockNb = (adderWidth / n).ceil();
+    final firstBlockSize = adderWidth - (blockNb - 1) * n;
+    final splitData = <int>[firstBlockSize];
+    for (var i = 1; i < blockNb; ++i) {
+      splitData.add(n);
+    }
+    return splitData;
+  }
+
+  /// Generator of splitter algorithm
+  static Splitter splitSelectAdderAlgorithmNBit(int n) =>
+      (adderWidth) => _splitAdderNBitFunctor(adderWidth, n);
+
+  /// Default adder functor to use.
+  static Adder defaultAdder(Logic a, Logic b,
+          {Logic? carryIn, Logic? subtractIn, String name = ''}) =>
+      ParallelPrefixAdder(a, b, carryIn: carryIn, name: name);
+
   /// Constructs a [CarrySelectCompoundAdder].
   CarrySelectCompoundAdder(super.a, super.b,
-      {Adder Function(Logic a, Logic b, {Logic? carryIn, String name})
-          adderGen = ParallelPrefixAdder.new,
+      {Adder Function(Logic a, Logic b,
+              {Logic? carryIn, Logic? subtractIn, String name})
```

Comment:
do all adders that work with this carry select adder have to support `subtractIn` for it to behave properly?

Reply:
This is a challenging API:   if the compound adder receives a subtractIn signal, then the adder generator must supply that signal.  But if it doesn't get one at its api, the sub-adders don't need it.